shell/browser/api/electron_api_app_mas.mm

```diff
@@ -34,7 +34,7 @@ void OnStopAccessingSecurityScopedResource(NSURL* bookmarkUrl) {
   NSError* error = nil;
   NSURL* bookmarkUrl =
       [NSURL URLByResolvingBookmarkData:bookmarkData
-                                options:NSURLBookmarkResolutionWithSecurityScope
+                                options:NSURLBookmarkResolutionWithSecurityScope|NSURLBookmarkResolutionWithoutUI|NSURLBookmarkResolutionWithoutMounting
```

Comment:
`WithoutMounting` makes sense to me here to address the specific problem being encountered. I don't know enough about this system to know if `WithoutUI` is desirable here.

cc @acheronfail who wrote this and @alexeykuzmin who reviewed it

Reply:
>`NSURLBookmarkResolutionWithoutUI`
Specifies that no UI feedback should accompany resolution of the bookmark data.

Seems totes reasonable that we want this as well.

Reply:
According to https://stackoverflow.com/a/2235076, that controls showing the auth dialog if the user needs to reauth, which maybe they do?

semi-related, http://michaellynn.github.io/2015/10/24/apples-bookmarkdata-exposed/ is a fascinating dive through some internals of this stuff.

Reply:
Seems like these should just be exposed as options `{ tryMountIfRequired, showUIIfRequired }`

Reply:
@MarshallOfSound in the interest of this being a fix rather than a feat, should we perhaps avoid changing the api surface?

Reply:
Fair, then we're just discussing whether to include the UI flag or not ❓ If the goal is completely silent bookmark usage then the no-ui flag makes sense IMO

Reply:
As I understand it, the goal is to stop hanging indefinitely when there are stuck network drives, and it's not clear to me that the UI flag is causing that problem.

Reply:
yeah I'm cool with removing the UI flag, it just seemed reasonable that this rando method should not be popping dialogs on our users either (although I'm yet to see a customer report with one)